packages/rum-core/src/domain/rumEventsCollection/view/trackViews.ts

```diff
@@ -115,15 +115,20 @@ export function trackViews(location: Location, lifeCycle: LifeCycle) {
     return { initialView, stop }
   }
 
-  function trackViewChange() {
-    return newView(lifeCycle, location, isRecording, ViewLoadingType.ROUTE_CHANGE, currentView.url)
+  function trackViewChange(startClocks?: ClocksState, name?: string) {
+    return newView(lifeCycle, location, isRecording, ViewLoadingType.ROUTE_CHANGE, currentView.url, startClocks, name)
   }
 
   return {
     addTiming: (name: string, endClocks = clocksNow()) => {
       currentView.addTiming(name, endClocks)
       currentView.triggerUpdate()
     },
+    startView: (name?: string, startClocks?: ClocksState) => {
+      currentView.end()
```

Comment:
The `end()` function will make the current view ends "now", but the newly started view could have a `startsClock` in the past, so the two views may overlap. I think we should use the `startsClock` value to set the end/duration of the ending view.